diff --git a/sound/core/seq/Makefile b/sound/core/seq/Makefile
index b29ffe835..b65fa5a19 100644
--- a/sound/core/seq/Makefile
+++ b/sound/core/seq/Makefile
@@ -7,7 +7,7 @@ snd-seq-device-objs := seq_device.o
 snd-seq-objs := seq.o seq_lock.o seq_clientmgr.o seq_memory.o seq_queue.o \
                 seq_fifo.o seq_prioq.o seq_timer.o \
                 seq_system.o seq_ports.o
-snd-seq-$(CONFIG_PROC_FS) += seq_info.o
+snd-seq-$(CONFIG_SND_PROC_FS) += seq_info.o
 snd-seq-midi-objs := seq_midi.o
 snd-seq-midi-emul-objs := seq_midi_emul.o
 snd-seq-midi-event-objs := seq_midi_event.o
